src/utils/components/ListPageFilter/ListPageFilter.tsx

```diff
@@ -0,0 +1,296 @@
+import React, { FC, useCallback, useState } from 'react';
+import { useDebounceCallback } from 'src/views/clusteroverview/utils/hooks/useDebounceCallback';
+
+import useDeepCompareMemoize from '@kubevirt-utils/hooks/useDeepCompareMemoize/useDeepCompareMemoize';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { FilterValue, OnFilterChange, RowFilter } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Badge,
+  Select,
+  SelectGroup,
+  SelectOption,
+  SelectVariant,
+  Toolbar,
+  ToolbarChip,
+  ToolbarContent,
+  ToolbarFilter,
+  ToolbarItem,
+  ToolbarToggleGroup,
+} from '@patternfly/react-core';
+import { FilterIcon } from '@patternfly/react-icons';
+
+import AutocompleteInput from './AutocompleteInput';
+import {
+  STATIC_SEARCH_FILTERS,
+  STATIC_SEARCH_FILTERS_LABELS,
+  STATIC_SEARCH_FILTERS_PLACEHOLDERS,
+} from './constants';
+import {
+  useQueryParamsMethods,
+  useRowFiltersParameters,
+  useSearchFiltersParameters,
+} from './hooks';
+import SearchFilter from './SearchFilter';
+import { Filter, FilterKeys, generateRowFilters, getFiltersData, intersection } from './utils';
+
+type ListPageFilterProps = {
+  data?: any;
+  loaded?: boolean;
+  onFilterChange?: OnFilterChange;
+  rowFilters?: RowFilter[];
+};
+
+const ListPageFilter: FC<ListPageFilterProps> = ({ data, loaded, onFilterChange, rowFilters }) => {
+  const { t } = useKubevirtTranslation();
+
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [isSelectOpen, setIsSelectOpen] = useState(false);
+
+  // Generate rowFilter items and counts. Memoize to minimize re-renders.
+  const generatedRowFilters = useDeepCompareMemoize(generateRowFilters(rowFilters ?? [], data));
+
+  // Reduce generatedRowFilters once and memoize
+  const [filters, filtersNameMap, filterKeys] = React.useMemo<
+    [Filter, FilterKeys, FilterKeys, string[]]
+  >(() => getFiltersData(generatedRowFilters), [generatedRowFilters]);
+
+  const { setOrRemoveQueryArgument } = useQueryParamsMethods();
+
+  const [selectedRowFilters, onRowFilterSearchParamChange] = useRowFiltersParameters({
+    filterKeys,
+    filters,
+  });
+
+  const textFilters = useSearchFiltersParameters();
+
+  const [searchType, setSearchType] = useState<keyof typeof STATIC_SEARCH_FILTERS>(
+    STATIC_SEARCH_FILTERS.name,
+  );
+  const [searchInputText, setSearchInputText] = useState(
+    searchType !== STATIC_SEARCH_FILTERS.labels ? textFilters[searchType] : '',
+  );
+
+  const onToggleClick = () => {
+    setIsDropdownOpen(!isDropdownOpen);
+  };
+
+  const onSelect = (event, value) => {
+    setSearchInputText(value !== STATIC_SEARCH_FILTERS.labels ? textFilters[value] : '');
```

Comment:
searchType !== STATIC_SEARCH_FILTERS.labels ? textFilters[searchType] : '' is already in the init sate. so maybe a func for it?